core/test/nuxeo-page-provider.test.js

```diff
@@ -108,4 +108,173 @@ suite('<nuxeo-page-provider>', () => {
       throw new Error('Expected to have an error response!');
     });
   });
+
+  suite('Parameters Transformation', () => {
+    function checkDefaultParameters(params, shouldIncludeKeys = [], expectedNumberOfNamedParameters = 0) {
+      const expectedKeys = ['currentPageIndex', 'namedParameters', 'offset', 'pageSize'];
+      const unexpectedKeys = ['query', 'queryParams', 'quickFilters', 'sortBy', 'sortOrder'];
+
+      shouldIncludeKeys.forEach((key) => {
+        if (unexpectedKeys.includes(key)) {
+          expectedKeys.push(key);
```

Comment:
Shouldn't `key` be added to expectedKeys even if it is not included in `unexpectedKeys `?

Reply:
It can be, but since we know and control the keys, it would just to enforce the test to fail.
I can understand the point and its advantages but I'm not sure if, at the end of the day, it is a better approach or not...